src/v/serde/parquet/column_writer.cc

```diff
@@ -66,16 +69,20 @@ class buffered_column_writer final : public column_writer::impl {
       , _max_def_level(schema_element.max_definition_level)
       , _opts(opts) {}
 
-    incremental_column_stats
-    add(value val, rep_level rl, def_level dl) override {
-        ++_num_values;
+    ss::future<> add(value val, rep_level rl, def_level dl) override {
         // A repetition level of zero means that it's the start of a new row and
         // not a repeated value within the same row.
         if (rl == rep_level(0)) {
+            // We can ONLY flush on row boundaries, so make sure this is the
+            // first thing we do.
+            if (_current_buffer_size > _opts.page_buffer_size) {
```

Comment:
wondering if this is better to at the end of add() (rather than at add time)

Reply:
We can't because we don't know when we end a row, only when we start one. We could move the batching policy up a level to the caller I suppose where we have more information

Reply:
ah yes, makes sense.